packages/syft/src/syft/service/worker/worker_pool_service.py

```diff
@@ -178,6 +179,56 @@ def delete_worker(
             message=f"Worker with id: {worker_id} deleted successfully from pool: {worker_pool.name}"
         )
 
+    @service_method(
+        path="worker_pool.get_worker",
+        name="get_worker",
+        roles=DATA_OWNER_ROLE_LEVEL,
+    )
+    def get_worker(
+        self, context: AuthedServiceContext, worker_pool_id: UID, worker_id: UID
+    ) -> Union[SyftWorker, SyftError]:
+        worker_pool_worker = self._get_worker_pool_and_worker(
+            context, worker_pool_id, worker_id
+        )
+        if isinstance(worker_pool_worker, SyftError):
+            return worker_pool_worker
+
+        worker_pool, worker = worker_pool_worker
+
+        worker_status = _get_worker_container_status(worker)
+        if isinstance(worker_status, SyftError):
+            return worker_status
+        elif worker_status != WorkerStatus.PENDING:
+            for worker in worker_pool.workers:
+                if worker.id == worker_id:
+                    worker.status = worker_status
+                    break
```

Comment:
```suggestion
            worker.status = worker_status
```

Reply:
Yeah this works, I thought I had to save the worker status back to the WorkerPool object before updating. Wasn't needed.